### PR TITLE
feat: [PL-38933]: Added FavoriteStar component

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

Added a component to mark an entity as favorite or un-favorite.  This component will be utilized in our upcoming Favorites feature.

figma - https://www.figma.com/file/H5adZ2kwJcyhM5U62AttOu/Favorites?type=design&node-id=803-189536&t=SipbJI3hYeJNXl4j-0

#### Screenshots

https://github.com/harness/harness-core-ui/assets/62028553/2f0cd38e-078f-4485-954d-1b0a17c48fc2

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by sunnykesh-harness — APPROVED** [review]

File: src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { useStrings } from 'framework/strings'
+import css from './FavoriteStar.module.scss'
+
+interface FavoriteStarProps {
+  resourceId: string
+  resourceType: CreateFavoriteAccountQueryParams['resource_type']
+  module?: Module
+  isFavorite?: boolean
+  projectIdentifier?: string
+  orgIdentifier?: string
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx:23]: a better interface could be:
```
resource: {
    id: string
    type: ...
    scope?: {
        projectIdentifier?: string
        orgIdentifier?: string
    }
}
```

File: src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    } finally {
+      setAPIInProgress(false)
+    }
+  }
+
+  const handleClick = (e: React.MouseEvent<Element, MouseEvent>): void => {
+    e.stopPropagation()
+
+    if (apiInProgress) {
+      showError(getString('common.previousActionInProgress'))
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx:93]: no need for a toast here.. just return silently

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx]: done

File: src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx (hunk 2 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/components/FavoriteStar/FavoriteStar.tsx]: done

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**